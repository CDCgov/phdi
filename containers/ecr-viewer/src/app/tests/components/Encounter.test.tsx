import { render } from "@testing-library/react";
import { axe } from "jest-axe";
import EncounterDetails from "../../view-data/components/Encounter";

describe("Encounter", () => {
  let container: HTMLElement;
  beforeAll(() => {
    const encounterData = [
      {
        title: "Encounter Type",
        value: "Ambulatory",
      },
      {
        title: "Encounter ID",
        value: "123456789",
      },
    ];
    const facilityData = [
      {
        title: "Facility ID",
        value: "2.16.840.1.113883.4.6",
      },
      {
        title: "Facility Name",
        value: "PRM- Palmdale Regional Medical Center",
      },
      {
        title: "Facility Type",
        value: "Healthcare Provider",
      },
    ];
    const providerData = [
      {
        title: "Provider Name",
        value: "test provider name",
      },
      {
        title: "Provider Contact",
        value: "test provider contact",
      },
    ];
    container = render(
      <EncounterDetails
        encounterData={encounterData}
        facilityData={facilityData}
        providerData={providerData}
      />,
    ).container;
  });
  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });
  it("should pass accessibility test", async () => {
    expect(await axe(container)).toHaveNoViolations();
  });
});
