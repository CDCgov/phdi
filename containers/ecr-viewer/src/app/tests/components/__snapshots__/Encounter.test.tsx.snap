// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encounter should match snapshot 1`] = `
<div>
  <div
    class="margin-top-0"
  >
    <div
      class="padding-bottom-3"
    >
      <div
        class="usa-summary-box__body"
      >
        <div>
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="encounter-details"
          >
            Encounter Details
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility ID
                </div>
                <div
                  class="grid-col maxw7 text-pre-line"
                >
                  2.16.840.1.113883.4.6
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line"
                >
                  PRM- Palmdale Regional Medical Center
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Facility Type
                </div>
                <div
                  class="grid-col maxw7 text-pre-line"
                >
                  Healthcare Provider
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
        <div
          class="margin-top-3"
        >
          <h4
            class="usa-summary-box__heading padding-y-105"
            id="provider-details"
          >
            Provider Details
          </h4>
          <div
            class="usa-summary-box__text"
          >
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Provider Name
                </div>
                <div
                  class="grid-col maxw7 text-pre-line"
                >
                  test provider name
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
            <div>
              <div
                class="grid-row"
              >
                <div
                  class="data-title padding-right-1"
                >
                  Provider Contact
                </div>
                <div
                  class="grid-col maxw7 text-pre-line"
                >
                  test provider contact
                </div>
              </div>
              <div
                class="section__line_gray"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
